app/src/main/java/ru/yandex/praktikumchatapp/presentation/ChatViewModel.kt

```diff
@@ -13,25 +14,26 @@ class ChatViewModel(
 
     private val repository = ChatRepository()
 
-    private val _messages = MutableLiveData<List<Message>>(emptyList())
-    val messages: LiveData<List<Message>> = _messages
+    private val _messages = MutableStateFlow(emptyList<Message>())
+    val messages = _messages.asStateFlow()
 
     init {
         viewModelScope.launch {
             while (isWithReplies) {
                 repository.getReplyMessage().collect { response ->
 
-                    val currentMessages = _messages.value ?: emptyList()
+                    val currentMessages = _messages.value
                     _messages.value =
                         currentMessages + Message.OtherMessage(response)
-
                 }
             }
         }
     }
 
     fun sendMyMessage(messageText: String) {
-        val currentMessages = _messages.value ?: emptyList()
-        _messages.value = currentMessages + Message.MyMessage(messageText)
+        _messages.update { messageList ->
+            val newListMessage: List<Message> = messageList + Message.MyMessage(messageText)
+            newListMessage
+        }
```

Comment:
Всю конструкцию можно написать в одну строчку, не потеряв читабельность:
```suggestion
        _messages.update { messageList -> messageList + Message.MyMessage(messageText)  }
```